### ISSUE TITLE
[Request]: Reinstate androidId - the unique Android hardware device ID.

### ISSUE THREAD

**Issue by davidpochon** [fluttercommunity/plus_plugins#954]
### Plugin

device_info_plus

### Use case

The **androidId** from the **AndroidDeviceInfo** class was a useful and in some cases essential means of _identifying a unique device_.

### Proposal

Used as a  persistent identifier, **androidId** should be reinstated in _device_info_plus_ having been dropped in version 4.0.0. 

Google themselves recognise the importance of this functionality. From the Android Developer [docs](https://support.google.com/googleplay/android-developer/answer/6048248#zippy=%2Cpersistent-identifiers-including-android-id):

> Persistent identifiers are still available because there are various supported use cases which aren’t related to advertising. Longer term, we’ll evaluate additional opportunities to provide users with even more informed control over what persistent identifiers are provided to third parties. We’ll communicate this prominently and in advance when we have details.

In short, _persistent identifiers_ are permitted provided the developer fulfils certain criteria. Therefore, it should be left to the developer to manage these criteria rather than dropping essential functionality that still exists on the underlying platform.

**Comment by Somtobro** [comment]
I totally agree with you

**Comment by naamapps** [comment]
100% agree with this.
Currently I'm using version 3.2.4 in order to use android I'd, but I'm concerned the version will be obsolete in future flutter versions and/or there won't be an alternative to get the androidId.

Please add the android ID functionality back to the package. It is a needed feature that a lot of developers use.

Thanks

**Comment by flechajm** [comment]
+1

**Comment by guidospadavecchia** [comment]
+1

**Comment by nicodumdum** [comment]
+1

**Comment by davidpochon** [comment]
For those interested, the issue that seemed to prompt the change is #824 and the pull request is #918. I have to agree with @feinstein that this [breaking change seems rushed](https://github.com/fluttercommunity/plus_plugins/issues/824#issuecomment-1168144384). The suggestion to use an older version of _device_info_plus_ is not a long term solution.

**Comment by vbuberen** [comment]
Check the reply in the issue about removing AndroidId: https://github.com/fluttercommunity/plus_plugins/issues/824#issuecomment-1199844679

**Comment by nohli** [comment]
We made a new plugin for getting the Android ID, so you can use the latest version of `device_info_plus` 🙂 
https://pub.dev/packages/android_id

**Comment by feinstein** [comment]
Nice! Thanks!

One improvement could be that since it's just a method in a class, it could have been made as a top level function. This follows the [dart style guide](https://dart.dev/guides/language/effective-dart/design#avoid-defining-a-one-member-abstract-class-when-a-simple-function-will-do).

**Comment by nohli** [comment]
> dart style guide

The linked rule is related to abstract classes and doesn't apply here, right?

Do you mean something like
```
Future<String?> getAndroidId() {
  const methodChannel = MethodChannel('android_id');
  final androidId = methodChannel.invokeMethod<String?>('getId');
  return androidId;
}
```

**Comment by feinstein** [comment]
Yeah, they talk about abstract classes, but I don't see much difference for a one method class. I can swear I remebem a text about one method classes, but maybe they updated the website.
